improve behavior in case of contradictory data (#130)
ZEISS/libczi: Src/libCZI/Doc/resolution_protocol.markdown

resolution protocol                 {#resolution_protocol_}
===================

# Resolution Protocol for Ambiguous or Contradictory Information

## Introduction

In cases where a CZI file contains ambiguous or contradictory information, the behavior
described in this document shall be considered canonical. Implementations MUST resolve
any conflicts according to these guidelines to ensure consistency and interoperability.  
Note that those guidelines only give recommendations how the data is to interpreted, this does not mean that
one should rely on this. 
A CZI containing mismatches (as discussed below) is without a doubt to be considered *malformed* and *corrupt*. However - if facing such
a malformed file, then a consistent behavior is desirable. Also, it is impractical to check internal consistency in all aspects before

> reading data from a CZI-file - it is designed in way that it allows to access only the parts of the file which are of interest (as opposed

Author: I have to confess that you suggestion gives me the idea that "access is only granted to those parts" (in the sense - everything else is forbidden). I'd like to leave as is for now.

Physically, the information is represented by the data-structure [SubBlockDirectoryEntryDV](https://github.com/ZEISS/libczi/blob/11015ae9aa97abbf9d78293a27115393077f9146/Src/libCZI/CziStructs.h#L175).
The information in this structure which is relevant for the payload content is:
* Pixel Type
* Compression
* physical width and height of the bitmap

It is now conceivable that the information in the DirectoryEntry is not consistent with the actual payload content. The following cases need to be considered:

1) If Compression is "uncompressed", then the expected size of the payload content can be calculated. In this case, a mismatch between the actual payload size and the expected size (calculated from pixel type and width and height) is possible.
2) If Compression is JPGXR, then the compressed data (in the payload) also contains information about the width and height of the bitmap. Or - when the payload content is decompressed, its width and height is determined (from the compressed data), and there might be a mismatch between this size and the width/height given in the DirectoryEntry.

> Note that in cases with Compression other than JPGXR, e.g. zstd0 or zstd1, it is not true that the width/height is reduntantly found in the compressed data. For zstd0/zstd1 compression, the decompression gives an unstructured blob of data, only described by its size. This case is therefore to be treated similar to case 1).

Reviewer: ```suggestion
Note that in cases with Compression other than JPGXR, e.g. zstd0 or zstd1, there is no redundant specification of the width/height in the compressed data. For zstd0/zstd1 compression, the decompression gives an unstructured blob of data, only described by its size. This case is therefore to be treated similar to case 1).
```

### case 1: payload-data size mismatch for uncompressed subblocks

The information in the DirectoyEntry is to be considered authoritative, i.e. the data is to be interpreted as a bitmap with width/height and pixel type as specified in the DirectoryEntry, and with a minimal stride.  

If the payload size is larger than what is calculated from width, height and pixel type, the additional data is to be discarded.   

> If the payload size is smaller than what is calculated from width, height and pixel type, then the missing data is to be considered containing zeroes.

Reviewer: ```suggestion
If the payload size is smaller than what is calculated from width, height and pixel type, then the missing data is to be filled with zeros.
```
